core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/model/CamelContextElement.java

```diff
@@ -38,6 +38,16 @@
 	 */
 	private Map<String, AbstractCamelModelElement> dataformats = new HashMap<>();
 	
+	/**
+	 * Contains REST configuration elements
+	 */
+	private Map<String, AbstractCamelModelElement> restConfigurations = new HashMap<>();
```

Comment:
The mentioned problem with the mandatory ids is coming from the implementation of the model which was based on the same pattern than for global elements, which are requiring ids.

Reply:
We are now generating IDs for the REST element and all REST operations (verbs)